### ISSUE TITLE
Flutter Local Notification not working on iOS with onesignal 

### ISSUE THREAD

**Issue by enadcs** [OneSignal/OneSignal-Flutter-SDK#591]
i already implement and used the pluign flutter_local_notifications and it was working fine , until i added the onesignal pluign , it stopped working on ios but working fine on android .

it seems that onesignal is blocking the application from displaying any local notification but its own notification or implementation for that 
 

(your description here)

**Environment**

flutter 3.0.1
ios 15.0
android 15

1. install flutter_local_notifications: ^9.0.0
2. install onesignal_flutter: ^3.1.0
3. try to display a local notification in any iOS device , it will not show anything

**Anything else:**

it show this error 

onesignalUserNotificationCenter:willPresentNotification:withCompletionHandler:

**Comment by emawby** [comment]
@enadcs Hello thank you for reaching out. It looks like this was reported a few years ago as well. Could you test to see if the resolution [here](https://github.com/OneSignal/OneSignal-Flutter-SDK/issues/197#issuecomment-590153195) helps still?

**Comment by muccy** [comment]
Pasting in AppDelegate
```
    override func userNotificationCenter(_ center: UNUserNotificationCenter, willPresent notification: UNNotification, withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
        super.userNotificationCenter(center, willPresent: notification, withCompletionHandler: completionHandler)
    }
    
    override func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
        super.userNotificationCenter(center, didReceive: response, withCompletionHandler: completionHandler)
    }
```

Seems to works. I have to do more tests to be sure

**Comment by Smartiiez** [comment]
@muccy thanks for the tip 👌

**Comment by MuhammadAbdullah7** [comment]
> Pasting in AppDelegate
> 
> ```
>     override func userNotificationCenter(_ center: UNUserNotificationCenter, willPresent notification: UNNotification, withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
>         super.userNotificationCenter(center, willPresent: notification, withCompletionHandler: completionHandler)
>     }
>     
>     override func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
>         super.userNotificationCenter(center, didReceive: response, withCompletionHandler: completionHandler)
>     }
> ```
> 
> Seems to works. I have to do more tests to be sure

what is the equivalent for appDelegate.swift

**Comment by FullstackWEB-developer** [comment]
Hi @muccy 
I tried it on latest version of Onesignal but it isn't working.

it's my AppDelegate.swift file:
```
import UIKit
import Flutter
import flutter_local_notifications

@main
@objc class AppDelegate: FlutterAppDelegate {
  override func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
  ) -> Bool {
    // This is required to make any communication available in the action isolate.
    FlutterLocalNotificationsPlugin.setPluginRegistrantCallback { (registry) in
      GeneratedPluginRegistrant.register(with: registry)
    }
    // if #available(iOS 10.0, *) {
    //   UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
    // }
    GeneratedPluginRegistrant.register(with: self)
    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
  }
  // for the OneSignal package
  override func userNotificationCenter(_ center: UNUserNotificationCenter, willPresent notification: UNNotification, withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
      super.userNotificationCenter(center, willPresent: notification, withCompletionHandler: completionHandler)
  }
  override func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
      super.userNotificationCenter(center, didReceive: response, withCompletionHandler: completionHandler)
  }
}
```

And this is my `flutter_local_notificatoins` helper singleton:
```

@pragma('vm:entry-point')
void notificationTapBackground(NotificationResponse notificationResponse) {
  // ignore: avoid_print
  print('notification(${notificationResponse.id}) action tapped: '
      '${notificationResponse.actionId} with'
      ' payload: ${notificationResponse.payload}');
  if (notificationResponse.input?.isNotEmpty ?? false) {
    // ignore: avoid_print
    print('notification action tapped with input: ${notificationResponse.input}');
  }
  FlutterLocalNotificationsHelper.onSelectNotification(notificationResponse);
}

class FlutterLocalNotificationsHelper {
  // prevent making instance
  FlutterLocalNotificationsHelper._();

  // Notification lib
  static FlutterLocalNotificationsPlugin flutterLocalNotificationsPlugin = FlutterLocalNotificationsPlugin();

  /// initialize local notifications service, create channels and groups
  /// setup notifications button actions handlers
  static init() async {
    // initialize local notifications
    await _initNotification();

    // request permission to show notifications
    _requestPermissions();
    // final NotificationAppLaunchDetails? notificationAppLaunchDetails =
    //     await flutterLocalNotificationsPlugin.getNotificationAppLaunchDetails();
    // if (notificationAppLaunchDetails?.didNotificationLaunchApp == true &&
    //     notificationAppLaunchDetails?.notificationResponse != null) {
    //   onSelectNotification(notificationAppLaunchDetails!.notificationResponse!);
    // }
  }

  /// request permission to show notifications
  static _requestPermissions() {
    flutterLocalNotificationsPlugin
        .resolvePlatformSpecificImplementation<IOSFlutterLocalNotificationsPlugin>()
        ?.requestPermissions(
          alert: true,
          badge: true,
          sound: true,
        );
    flutterLocalNotificationsPlugin
        .resolvePlatformSpecificImplementation<AndroidFlutterLocalNotificationsPlugin>()
        ?.requestNotificationsPermission();
  }

  ///init notifications channels
  static _initNotification() async {
    const initializationSettingsAndroid = AndroidInitializationSettings('ic_launcher');

    const initializationSettingsIOS = DarwinInitializationSettings(
      onDidReceiveLocalNotification: onDidReceiveLocalNotification,
      // notificationCategories: darwinNotificationCategories,
    );

    const initializationSettings = InitializationSettings(
      android: initializationSettingsAndroid,
      iOS: initializationSettingsIOS,
    );

    await flutterLocalNotificationsPlugin.initialize(
      initializationSettings,
      onDidReceiveNotificationResponse: onSelectNotification,
    );

    // Define notification channels
    const AndroidNotificationChannel generalChannel = AndroidNotificationChannel(
      'general_channel', // id
      'General Notifications', // name
      description: 'Notification channel for general notifications',
      importance: Importance.max,
    );

    // Create the notification channels
    await flutterLocalNotificationsPlugin
        .resolvePlatformSpecificImplementation<AndroidFlutterLocalNotificationsPlugin>()
        ?.createNotificationChannel(generalChannel);
  }

  //display notification for user with sound
  static showNotification(
    String title,
    String body,
    String id,
    String? link,
  ) async {
    const AndroidNotificationDetails androidPlatformChannelSpecifics = AndroidNotificationDetails(
      'general_channel', // channel id
      'General Notifications', // channel name
      channelDescription: 'Notification channel for general notifications',
      importance: Importance.max,
      priority: Priority.high,
      ticker: 'ticker'
      // showWhen: true,
    );

    // TODO: FIX Implement iOS notification details
    const DarwinNotificationDetails iOSPlatformChannelSpecifics =
        DarwinNotificationDetails();

    const NotificationDetails platformChannelSpecifics = NotificationDetails(
      android: androidPlatformChannelSpecifics,
      iOS: iOSPlatformChannelSpecifics,
    );

    await flutterLocalNotificationsPlugin.show(
      id.hashCode,
      title,
      body,
      platformChannelSpecifics,
      payload: jsonEncode({'id': id, 'link': link}), // Use a Map to pass additional data
    );
  }

  /// Handle notification tapped logic here
  static Future onSelectNotification(NotificationResponse notificationResponse) async {
    debugPrint('notification tapped: ${notificationResponse.payload}');
  }

  /// Handle iOS local notification received
  static Future onDidReceiveLocalNotification(int id, String? title, String? body, String? payload) async {
    // Handle iOS local notification received
    debugPrint('iOS local notification received with id: $id, title: $title, body: $body, payload: $payload');
  }
}
```

And I tried to click this button:
```
ElevatedButton(
  onPressed: () async {
    FlutterLocalNotificationsHelper.showNotification(
      'title',
      'body',
      "0",
      'https://www.google.com',
    );
  },
  child: Text('Show plain notification with payload'),
),
```

**Comment by luis901101** [comment]
Hi @FullstackWEB-developer 

Uncomment this in your `AppDelegate.swift`:

```swift
if #available(iOS 10.0, *) {
  UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
}
```

**Comment by fimpact-nicole** [comment]
> Hi @FullstackWEB-developer
> 
> Uncomment this in your `AppDelegate.swift`:
> 
> ```swift
> if #available(iOS 10.0, *) {
>   UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
> }
> ```

@luis901101 Thank you so much!
